ui/store/actions.ts

```diff
@@ -3282,6 +3282,45 @@ export function setParticipateInMetaMetrics(
   };
 }
 
+export function setDataCollectionForMarketing(
+  dataCollectionPreference: boolean,
+): ThunkAction<
+  Promise<[boolean, string]>,
+  MetaMaskReduxState,
+  unknown,
+  AnyAction
+> {
+  return (dispatch: MetaMaskReduxDispatch) => {
+    log.debug(`background.setDataCollectionForMarketing`);
+    return new Promise((resolve, reject) => {
+      callBackgroundMethod<string>(
+        'setDataCollectionForMarketing',
+        [dataCollectionPreference],
+        (err, metaMetricsId) => {
+          log.debug(err);
+          if (err) {
+            dispatch(displayWarning(err));
+            reject(err);
+            return;
+          }
+          /**
+           * We need to inform sentry that the user's optin preference may have
+           * changed. The logic to determine which way to toggle is in the
+           * toggleSession handler in setupSentry.js.
+           */
+          window.sentry?.toggleSession();
```

Comment:
This doesn't seem necessary; this toggle doesn't impact Sentry. Only the overall meta metrics toggle impacts Sentry.

Reply:
Fixed